ui_components/src/chat/Chat_new.js

```diff
@@ -27,20 +28,23 @@ type Props = {|
 function Chat({
     chatApiLink,
     initialPosts,
+    initialPostId,
     initialNextPagePostId,
+    initialPreviousPagePostId,
     initialLatestPostId,
     maxPostLength,
     isModerator,
     initialBanInfo,
-    memes
+    memes,
 }: Props) {
     const [banInfo, setBanInfo] = React.useState(initialBanInfo);
     const [posts, setPosts] = React.useState(initialPosts);
 
     const [nextPagePostId, setNextPagePostId] = React.useState(initialNextPagePostId);
     const [latestPostId, setLatestPostId] = React.useState(initialLatestPostId);
     // Only set if we're paginating
-    const [previousPagePostId, setPreviousPagePostId] = React.useState(null);
+    const [previousPagePostId, setPreviousPagePostId] = React.useState(initialPreviousPagePostId);
+    const initialPostIdRef = React.useRef(initialPostId);
```

Comment:
Hmm is this doing something functionally different than `currentPagePostIdRef`, or could we initialize `currentPagePostIdRef` to that value and drop this var completely?

Reply:
`initialPostIdRef` gives us the option to target the requested post for things like highlighting.

Also if we initialize `currentPagePostIdRef` with the current logic then `refreshChat` won't run when you view latest results.

```
const refreshChat = function() {
        if(currentPagePostIdRef.current != null) {
            return;
        }```

Reply:
> initialPostIdRef gives us the option to target the requested post for things like highlighting.
Makes sense. I think what we could do to make this clearer is create a state variable called `highlightPostId` and initialize it to the value of `initialPostId`. 

> Also if we initialize currentPagePostIdRef with the current logic then refreshChat won't run when you view latest results.
Calling `changePage(latestPostId)` will actually clear the `currentPagePostIdRef`, same way as continually clicking "newer" until you hit the latest post clears it